IdentityCore/src/network/MSIDHttpRequest.m

```diff
@@ -66,7 +67,19 @@ - (void)sendWithBlock:(MSIDHttpRequestDidCompleteBlock)completionBlock
     requestConfigurator.timeoutInterval = _requestTimeoutInterval;
     [requestConfigurator configure:self];
     
-    self.urlRequest = [self.requestSerializer serializeWithRequest:self.urlRequest parameters:self.parameters];
+    self.urlRequest = [self.requestSerializer serializeWithRequest:self.urlRequest parameters:self.parameters headers:self.headers];
+    NSCachedURLResponse *response = [self cachedResponse];
+    if (response)
+    {
+        NSError *error = nil;
+        id responseObject = [self.responseSerializer responseObjectForResponse:(NSHTTPURLResponse *)response.response
+                                                                          data:response.data
+                                                                       context:self.context
+                                                                         error:&error];
+                 
+        if (completionBlock) { completionBlock(responseObject, error); }
```

Comment:
I think here you should discard cached object and go to network if response serialization fails (if response serializer returns no object for a cached response or error). 
For example, imagine a situation where device is on a captive network. Instead of authority metadata you get an html page that your URL cache caches (since you got 200 OK). Now every time when client is trying to re-download authority metadata, parsing will always fail.

Reply:
Addressed. Just log the error in this case instead of creating an error.